tensorflow/tools/ci_build/builds/cmake.sh

```diff
@@ -16,7 +16,53 @@
 
 set -e
 
+
+# Determine the number of cores, for parallel make.
+N_JOBS=$(grep -c ^processor /proc/cpuinfo)
+if [[ -z ${N_JOBS} ]]; then
+  # The Linux way didn't work. Try the Mac way.
+  N_JOBS=$(sysctl -n hw.ncpu)
+fi
+if [[ -z ${N_JOBS} ]]; then
+  N_JOBS=1
+  echo ""
+  echo "WARNING: Failed to determine the number of CPU cores. "\
+"Will use --jobs=1 for make."
+fi
+
+echo ""
+echo "make will use ${N_JOBS} concurrent job(s)."
+echo ""
+
+
+# Build and install protobuf.
```

Comment:
Would it be worth factoring this out into a separate install script? AIUI (and my Docker knowledge is limited), this will cause every CMake build to download and build protobuf again; and if we moved it into an install script run by `Dockerfile.cmake`, the image would be cached between runs. Or is there a downside to such a move?